### PR TITLE
Implement geofence 

### PR DESCRIPTION
**Pull request by cp-ishita-g**
- [x] create service handler file for geofence
- [x] create repository file for geofence
- [x] init geofence in main
- [x] create space repository to get space list only in that file
- [x] make http callable for send geofence notification
- [x] test geofence notification
- [x] handle tap for geofence push notification

https://github.com/user-attachments/assets/17a11d17-065e-44c2-870f-9e5c954eab2f

### CODE REVIEW

**Comment by kaushiksaliya** [comment]
why this location prompt is showing in home screen when app is launch.

[Fix prompt.webm](https://github.com/user-attachments/assets/55ea17ff-b812-4cba-93c3-84915814b98c)

File: app/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
-    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
-    <uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION"/>
     <uses-permission android:name="android.permission.POST_NOTIFICATIONS"/>
     <uses-permission android:name="android.permission.REQUEST_IGNORE_BATTERY_OPTIMIZATIONS" />
     <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
+    <uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION"/>
+    <uses-permission android:name="android.permission.WAKE_LOCK" />
+    <uses-permission android:name="com.google.android.gms.permission.ACTIVITY_RECOGNITION" />
```
[the remark this answers is not in the document]
Comment (cp-ishita-g) [reply app/android/app/src/main/AndroidManifest.xml:11]: it was given in https://pub.dev/packages/geofence_service to add this permission

**Comment by cp-ishita-g** [comment]
> why this location prompt is showing in home screen when app is launch.

because geofence track with device `Geofence can see what activity took place when the device entered the radius`, that's why and geofence will start register in at where we have initialisation

**Comment by kaushiksaliya** [comment]
> > why this location prompt is showing in home screen when app is launch.
> 
> because geofence track with device `Geofence can see what activity took place when the device entered the radius`, that's why and geofence will start register in at where we have initialisation

removed this we already asking permission in side the permission screen.

**Comment by kaushiksaliya** [comment]
add notification testing video

**Review by kaushiksaliya — APPROVED** [review]

**Pull request merged**